File: Practice/src/coding_basic_training/ThreeDelimiters.py
def solution(myStr):
    answer = []
    l = 0
    r = 0
    while r <= len(myStr):
        if r == len(myStr):
            if l < r:
                answer.append((myStr[l:r]))
            break;
        if myStr[r] == 'a' or myStr[r] == 'b' or myStr[r] == 'c':
            if l < r:
                answer.append(myStr[l:r])
            l = r + 1
        r += 1
    return answer if answer else ["EMPTY"]
    # answer = myStr.replace("a", " ").replace("b", " ").replace("c", " ").split()
    # return answer if answer else ["EMPTY"]

File: Practice/src/coding_basic_training/ThreeDelimiters.java
package coding_basic_training;

import java.util.ArrayList;

public class ThreeDelimiters {
//	세 개의 구분자
	public String[] solution(String myStr) {
		ArrayList<String> li = new ArrayList<String>();
		int l = 0;
		int r = 0;
		while (r < myStr.length()) {
			char c = myStr.charAt(r);
			if (c == 'a' || c == 'b' || c == 'c') {
				if (l < r) {
					li.add(myStr.substring(l, r));
				}
				l = r + 1;
			}
			r++;
		}
		if (l < r) {
			li.add(myStr.substring(l, r));
		}
		return li.size() > 0 ? li.toArray(String[]::new) : new String[] { "EMPTY" };
	}
}